cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginClass.java

```diff
@@ -131,6 +149,7 @@ public int hashCode() {
     result = 31 * result + className.hashCode();
     result = 31 * result + (configFieldName != null ? configFieldName.hashCode() : 0);
     result = 31 * result + properties.hashCode();
+    result = 31 * result + (endpoints != null ? endpoints.hashCode() : 0);
```

Comment:
you need to modify equals() too.  While doing it, can switch to use java.util.Objects, which would make it slightly easier to modify in the future